components/brave_new_tab_ui/hooks/PrefHookManager.ts

```diff
@@ -122,7 +122,7 @@ export function createPrefsHook<OnType extends {}>(prefManager: PrefHookManager<
         }, [prefManager, pref])
 
         return [
-            prefManager.getPref(pref),
+            value,
```

Comment:
@fallaciousreasoning what do you think about this fix? i noticed that effects dependent on a prefs hook would not react to changes without this fix.

Reply:
Seems reasonable enough to me :smile: Not sure why effects dependent on a prefs hook aren't reacting though - that seems super weird.

Reply:
Actually looking at https://github.com/brave/brave-core/pull/24141/files#r1635640377 I changed it because there was some sort of caching issue.

Might be worth a deeper look

Reply:
>I changed it because there was some sort of caching issue.

I believe I saw something similar when using the search widget toggle in the settings panel, as the toggle state would not change upon the first click. adding `setValue` to the `setPref` callback seemed to fix the issue

https://github.com/brave/brave-core/blob/0dad485415ee174a492980e14d36f2e5aee6d17c/components/brave_new_tab_ui/hooks/PrefHookManager.ts#L110-L113